### DISCUSSION TITLE
Async Storage Encryption

### DISCUSSION THREAD

**Discussion by Maharshi01** [react-native-async-storage/async-storage#590] [Q&A] [answered]
Is it possible to encrypt async storage? Is there a plan to add secure Storage for Async Storage?
Current version does not include any encryption selection..

**Comment by tido64** [comment, marked as answer]
As far as I know, there are no plans to add encryption to AsyncStorage. @Krizzu correct me if I'm wrong. However, there's nothing preventing you from encrypting the data before handing it over to AsyncStorage. There seems to be plenty of packages providing this feature: https://www.npmjs.com/search?q=keywords:crypto

**Reply by Maharshi01** [reply]
@tido64 Thanks for replying. I have a question though, will there be any impact in performance to the app if i encrypt and store using one of the encryption package.

**Reply by tido64** [reply]
It will definitely have some impact since you'll be doing extra work, but how much will depend on your workload. You'll most likely need to do some benchmarking yourself. I suppose you could also ask the package authors for some numbers to give you an idea of what it might be, but it won't be accurate for your app.

**Reply by Maharshi01** [reply]
@tido64 Thanks, Do you mean crypto packages authors ?

**Reply by tido64** [reply]
Yes.

**Reply by Maharshi01** [reply]
Ok, Thanks @tido64
